### PR TITLE
Replace flake8 with ruff 

### PR DESCRIPTION
**Pull request by CoMPaTech**
Following up on plugwise-beta 367

As for following core, ruff is executed through pre-commit.ci so it's run outside of the GitHub actions

### CODE REVIEW

**Comment by CoMPaTech** [comment]
For review purposes [1283d53](https://github.com/plugwise/python-plugwise/pull/279/commits/1283d539a813dc6f7f8add834cdedef0cd845d4e) is mostly the descriptions changing (no code except the removal of the imports in __init__)? the two commits following are about code changes (where I guess we should review if we can follow the suggestion of `helper.py` L797)

**Review by bouwew — APPROVED** [review]
LGTM!

**Comment by bouwew** [comment]
On L797, what happens when you remove the "'s around "icon"?
The tests still pass? If yes, then OK to remove, just check if the test-coverage is there/correct.

**Comment by CoMPaTech** [comment]
> On L797, what happens when you remove the "'s around "icon"?
> The tests still pass? If yes, then OK to remove, just check if the test-coverage is there/correct.

icon is a string not an existing object - this fugly construct works but we should be able to improve it, code wise it works

**Review by bouwew — APPROVED** [review]
LGTM

Let's merge this one before #281

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -39,20 +39,12 @@ repos:
     hooks:
       - id: codespell
         exclude_types: [csv, json]
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
```
Comment (frenck) [inline .pre-commit-config.yaml:42]: You want to run this before isort & black.

If it fixes thing, it will be formatted after that...

Comment (bouwew) [reply .pre-commit-config.yaml:42]: Ok thanks, will put it in a separate PR, I just merged this one :)

Comment (bouwew) [reply .pre-commit-config.yaml:42]: https://github.com/plugwise/python-plugwise/pull/282

**Pull request merged**